src/main/java/uk/gov/ons/ctp/integration/contactcentresvc/service/impl/CaseServiceImpl.java

```diff
@@ -1059,4 +1057,54 @@ private void sendEvent(EventType eventType, EventPayload payload, Object caseId)
         .with("transactionId", transactionId)
         .debug("{} event published", eventType);
   }
+
+  private CaseDTO getLatestCaseById(UUID caseId, Boolean getCaseEvents) throws CTPException {
+
+    List<CaseDTO> cases = new ArrayList<>();
+    CaseContainerDTO caseFromRM = null;
+    CaseDTO caseDto = null;
+
+    try {
+      caseFromRM = getCaseFromRm(caseId, getCaseEvents);
+    } catch (ResponseStatusException ex) {
+      if (ex.getStatus() == HttpStatus.NOT_FOUND) {
+        log.with("caseId", caseId).debug("Case Id Not Found by Case Service");
+      } else {
+        log.with("caseId", caseId)
+            .with("status", ex.getStatus())
+            .error("Error calling Case Service");
+        throw ex;
+      }
+    }
+
+    if (caseFromRM != null) {
+      caseDto = mapCaseContainerDTO(caseFromRM);
+      cases.add(caseDto);
+    }
+
+    List<CaseDTO> cachedCases =
+        dataRepo
+            .readCachedCaseById(caseId)
+            .stream()
+            .map(this::createNewCachedCaseResponse)
+            .collect(toList());
+    log.with("caseId", caseId)
+        .with("cases", cachedCases.size())
+        .debug("Found {} case details in Cache for caseId", cachedCases.size());
+
+    TimeOrderedCases timeOrderedCases = new TimeOrderedCases();
+    timeOrderedCases.add(cases);
+    timeOrderedCases.add(cachedCases);
+    Optional<CaseDTO> latest = timeOrderedCases.latest();
+
+    CaseDTO latestCaseDto = null;
+    if (latest.isPresent()) {
+      latestCaseDto = latest.get();
+    } else {
+      log.with("caseId", caseId).warn("Request for case Not Found");
```

Comment:
No code coverage here - add a unit test

Reply:
I've now added two:
testRMCaseNotFound()
testRMAndCachedCaseNotFound()

But I'm not sure that the testRMCaseNotFound() one is needed so I can remove it if you like?

Reply:
I'll need to see exactly what the test cases do but I think that the code needs to be able to handle 3 different circumstances:

- An unknown case. Not in cache or RM
- Case in cache, but not yet reached RM
- Case in cache & RM